posthog/test/test_feature_flag.py

```diff
@@ -57,9 +58,104 @@ def test_complicated_flag(self):
             }
         )
 
-        self.assertEqual(FeatureFlagMatcher(feature_flag, "test_id").get_match(), FeatureFlagMatch())
-        self.assertEqual(FeatureFlagMatcher(feature_flag, "example_id").get_match(), FeatureFlagMatch())
-        self.assertIsNone(FeatureFlagMatcher(feature_flag, "another_id").get_match())
+        self.assertEqual(FeatureFlagMatcher([feature_flag], "test_id").get_match(feature_flag), FeatureFlagMatch())
+        self.assertEqual(FeatureFlagMatcher([feature_flag], "example_id").get_match(feature_flag), FeatureFlagMatch())
+        self.assertIsNone(FeatureFlagMatcher([feature_flag], "another_id").get_match(feature_flag))
+
+    @snapshot_postgres_queries
+    def test_multiple_flags(self):
+        Person.objects.create(
+            team=self.team, distinct_ids=["test_id"], properties={"email": "test@posthog.com"},
+        )
+        self.create_groups()
+        feature_flag_one = self.create_feature_flag(
+            filters={
+                "groups": [
+                    {
+                        "properties": [
+                            {"key": "email", "type": "person", "value": "test@posthog.com", "operator": "exact"}
+                        ],
+                        "rollout_percentage": 100,
+                    },
+                    {"rollout_percentage": 50},
+                ]
+            },
+            key="one",
+        )
+        feature_flag_always_match = self.create_feature_flag(
+            filters={"groups": [{"rollout_percentage": 100}]}, key="always_match"
+        )
+        feature_flag_never_match = self.create_feature_flag(
+            filters={"groups": [{"rollout_percentage": 0}]}, key="never_match"
+        )
+        feature_flag_group_match = self.create_feature_flag(
+            filters={"aggregation_group_type_index": 1, "groups": [{"rollout_percentage": 100}],}, key="group_match"
+        )
+        feature_flag_group_no_match = self.create_feature_flag(
+            filters={"aggregation_group_type_index": 1, "groups": [{"rollout_percentage": 0}],}, key="group_no_match"
+        )
+        feature_flag_group_property_match = self.create_feature_flag(
+            filters={
+                "aggregation_group_type_index": 0,
+                "groups": [
+                    {
+                        "rollout_percentage": 100,
+                        "properties": [
```

Comment:
@neilkakkar This was very subtle but you used the old way of setting properties (just a dict, no array) which caused Filter to interpret these as event properties rather than group properties.

Reply:
facepalming hardd, right, the sql looks good now!